pkg/k8s/resource/resource.go

```diff
@@ -532,6 +543,25 @@ func (r *resource[T]) Events(ctx context.Context, opts ...EventsOpt) <-chan Even
 	return out
 }
 
+func (r *resource[T]) resourceName() string {
+	if r.opts.name != "" {
+		return r.opts.name
+	}
+
+	// We create a new pointer to the reconciled resource type.
+	// For example, with resource[*cilium_api_v2.CiliumNode] new(T) returns **cilium_api_v2.CiliumNode
+	// and *new(T) is nil. So we create a new pointer using reflect.New()
+	o := *new(T)
+	sourceObj := reflect.New(reflect.TypeOf(o).Elem()).Interface().(T)
```

Comment:
I'm not sure I understand what would happen if we don't have this code.

Reply:
Without this code, we won't get the GVK for the reconciled object. GVKForObject finds the GVK associated with the given object and if we pass a nil pointer it can't find it.

https://github.com/kubernetes/apimachinery/blob/3e2600dc79feea6cdc8a9224bc8a6a7fcfee1466/pkg/runtime/scheme.go#L243